File: tests/bench/lua/prime_sieve.lua
function sieve(n)
    local count = 0
    for i = 2, n - 1 do
        local is_prime = true
        local j = 2
        while j * j <= i do
            if i % j == 0 then
                is_prime = false
                break
            end
            j = j + 1
        end
        if is_prime then count = count + 1 end
    end
    return count
end
print(sieve(5000))

File: tests/bench/php/prime_sieve.php
<?php
function sieve($n) {
    $count = 0;
    for ($i = 2; $i < $n; $i++) {
        $is_prime = true;
        for ($j = 2; $j * $j <= $i; $j++) {
            if ($i % $j == 0) { $is_prime = false; break; }
        }
        if ($is_prime) $count++;
    }
    return $count;
}
echo sieve(5000) . "\n";
